### PR TITLE
ML Streaming docstrings

### PR DESCRIPTION
**Pull request by AHReccese**
#### Reference Issues/PRs
Add docstrings to the streaming module

### CODE REVIEW

File: pymilo/streaming/communicator.py [hunk 1]
```diff
@@ -91,11 +135,19 @@ async def attribute_call(request: Request):
             }
 
     def parse(self, body):
+        """
+        Parse the compressed encrypted body of the request.
+
+        :param body: request body
+        :type body: str
+        :return: str (the extracted decrypted version)
```
Comment (sadrasabouri) [inline pymilo/streaming/communicator.py:143]: `:return: the extracted decrypted version`

File: pymilo/streaming/compressor.py [hunk 2]
```diff
@@ -1,12 +1,17 @@
+# -*- coding: utf-8 -*-
+"""Implementations of Compressor interface."""
 from .interfaces import Compressor
 
 
 class DummyCompressor(Compressor):
+    """The Pymilo DummyCompressor class is a dummy implementation of the Compressor interface to act as a simple wire."""
```
Comment (sadrasabouri) [inline pymilo/streaming/compressor.py:7]:
```
A dummy implementation of the Compressor interface.
```

File: pymilo/streaming/encryptor.py [hunk 3]
```diff
@@ -1,12 +1,17 @@
+# -*- coding: utf-8 -*-
+"""Implementations of Encryptor interface."""
 from .interfaces import Encryptor
 
 
 class DummyEncryptor(Encryptor):
+    """The Pymilo DummyEncryptor class is a dummy implementation of the Encryptor interface to act as a simple wire."""
```
Comment (sadrasabouri) [inline pymilo/streaming/encryptor.py:7]: same

File: pymilo/streaming/interfaces.py [hunk 4]
```diff
@@ -4,44 +4,94 @@
...
+    """
 
     @abstractmethod
     def compress(payload):
         """
+        Compress the given payload.
+
+        :param payload: payload to get compressed
+        :type payload: str
+        :return: str (compressed payload)
```
Comment (sadrasabouri) [inline pymilo/streaming/interfaces.py:20]: same

File: pymilo/streaming/interfaces.py [hunk 5]
```diff
@@ -4,44 +4,94 @@
...
         """
 
     @abstractmethod
     def extract(payload):
         """
+        Extract the given previously compressed payload.
+
+        :param payload: payload to get extracted
+        :type payload: str
+        :return: str (extracted payload)
```
Comment (sadrasabouri) [inline pymilo/streaming/interfaces.py:30]: same

File: pymilo/streaming/interfaces.py [hunk 6]
```diff
@@ -4,44 +4,94 @@
...
+    """
 
     @abstractmethod
     def encrypt(payload):
         """
+        Encrypt the given payload.
+
+        :param payload: payload to get encrypted
+        :type payload: str
+        :return: str (encrypted payload)
```
Comment (sadrasabouri) [inline pymilo/streaming/interfaces.py:48]: same

File: pymilo/streaming/interfaces.py [hunk 7]
```diff
@@ -4,44 +4,94 @@
...
         """
 
     @abstractmethod
     def decrypt(payload):
         """
+        Decrypt the given previously encrypted payload.
+
+        :param payload: payload to get decrypted
+        :type payload: str
+        :return: str (decrypted payload)
```
Comment (sadrasabouri) [inline pymilo/streaming/interfaces.py:58]: same

File: pymilo/streaming/interfaces.py [hunk 8]
```diff
@@ -4,44 +4,94 @@
...
+    """
 
     @abstractmethod
     def upload(self, payload):
         """
+        Upload the given payload to the remote server.
+
+        :param payload: request payload
+        :type payload: dict
+        :return: Response object (varies based on the Implemented Protocol)
```
Comment (sadrasabouri) [inline pymilo/streaming/interfaces.py:76]: same

File: pymilo/streaming/interfaces.py [hunk 9]
```diff
@@ -4,44 +4,94 @@
...
         """
 
     @abstractmethod
     def download(self, payload):
         """
+        Download the remote ML model to local.
+
+        :param payload: request payload
+        :type payload: dict
+        :return: Response object (varies based on the Implemented Protocol)
```
Comment (sadrasabouri) [inline pymilo/streaming/interfaces.py:86]: same

File: pymilo/streaming/interfaces.py [hunk 10]
```diff
@@ -4,44 +4,94 @@
...
         """
 
     @abstractmethod
     def attribute_call(self, payload):
         """
+        Execute an attribute call on the remote server.
+
+        :param payload: request payload
+        :type payload: dict
+        :return: Response object (varies based on the Implemented Protocol)
```
Comment (sadrasabouri) [inline pymilo/streaming/interfaces.py:96]: same

File: pymilo/streaming/pymilo_client.py [hunk 11]
```diff
@@ -1,9 +1,12 @@
+# -*- coding: utf-8 -*-
+"""PyMiloClient for RESTFull protocol."""
```
Comment (sadrasabouri) [inline pymilo/streaming/pymilo_client.py:2]: `Protocol`

File: pymilo/streaming/pymilo_client.py [hunk 12]
```diff
@@ -1,9 +1,12 @@
...
 from .encryptor import DummyEncryptor
 from .compressor import DummyCompressor
 from ..pymilo_obj import Export, Import
 from .communicator import RESTClientCommunicator
 from ..transporters.general_data_structure_transporter import GeneralDataStructureTransporter
 class PymiloClient:
+    """The Pymilo PymiloClient class facilitates working with PyMiloServer."""
```
Comment (sadrasabouri) [inline pymilo/streaming/pymilo_client.py:9]: same

File: pymilo/streaming/pymilo_server.py [hunk 13]
```diff
@@ -1,3 +1,5 @@
+# -*- coding: utf-8 -*-
+"""PyMiloServer for RESTFull protocol."""
```
Comment (sadrasabouri) [inline pymilo/streaming/pymilo_server.py:2]: same

File: pymilo/streaming/pymilo_server.py [hunk 14]
```diff
@@ -6,21 +8,45 @@
 
 
 class PymiloServer:
+    """The Pymilo PymiloServer class facilitates streaming the ML models."""
```
Comment (sadrasabouri) [inline pymilo/streaming/pymilo_server.py:11]: same

File: tests/test_ml_streaming/run_server.py [hunk 15]
```diff
@@ -1,3 +1,4 @@
 from pymilo.streaming.pymilo_server import PymiloServer
 
-fastapi_app = PymiloServer()._communicator.app
+communicator = PymiloServer()._communicator
+communicator.run()
```
Comment (sadrasabouri) [inline tests/test_ml_streaming/run_server.py:4]: Add an extra line at the end of this file.

**Review by sadrasabouri — CHANGES_REQUESTED** [review]
Thanks for your efforts. I left some comments.

**Comment by AHReccese** [comment]
@sadrasabouri Hi Sadra, Thank you for your comprehensive review, I will apply them all in the coming 4-5 hours.

**Review by sadrasabouri — APPROVED** [review]

**Pull request merged**